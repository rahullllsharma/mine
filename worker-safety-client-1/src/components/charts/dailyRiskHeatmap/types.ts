/* eslint-disable @typescript-eslint/no-explicit-any */
import type { RiskLevel } from "@/components/riskBadge/RiskLevel";

export type RiskLevelByDate = {
  date: string;
  riskLevel: RiskLevel;
};

export type ProjectRiskLevelByDate = {
  projectName: string;
  riskLevelByDate: RiskLevelByDate[];
  dateToRiskLevel?: DateToRiskLevel;
};

export type LocationRiskLevelByDate = {
  locationName: string;
  riskLevelByDate: RiskLevelByDate[];
  dateToRiskLevel?: DateToRiskLevel;
};

export type TaskRiskLevelByDate = {
  taskName: string;
  locationName: string;
  projectName: string;
  riskLevelByDate: RiskLevelByDate[];
  dateToRiskLevel?: DateToRiskLevel;
};

export type EntityRiskLevelByDate =
  | ProjectRiskLevelByDate
  | LocationRiskLevelByDate
  | TaskRiskLevelByDate;

export type DateToRiskLevel = { [date: string]: RiskLevel };

export type ColumnDef = {
  id: string;
  Header: string;
  width?: number;
  value?: (row: any) => string;
  accessor?: (entityRisk: EntityRiskLevelByDate, index?: number) => any;
};
