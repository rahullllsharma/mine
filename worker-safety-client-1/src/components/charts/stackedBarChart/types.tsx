import type { ComputedDatum, BarDatum } from "@nivo/bar";

export type StackedBarChartDataDescription = {
  key: string;
  color?: string | ((datum: ComputedDatum<BarDatum>) => string);
  hoverColor?: string | ((datum: ComputedDatum<BarDatum>) => string);
  selectedColor?: string | ((datum: ComputedDatum<BarDatum>) => string);
  labelColor?: string | ((datum: ComputedDatum<BarDatum>) => string);
};
