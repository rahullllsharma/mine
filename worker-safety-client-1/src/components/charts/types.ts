import type { RiskLevel } from "@/components/riskBadge/RiskLevel";

// TODO perhaps these types are derived from the graphql via apollo?

export interface LocationRiskCount {
  date: string;
  riskLevel: RiskLevel;
  count: number;
}

export interface ProjectRiskCount {
  date: string;
  riskLevel: RiskLevel;
  count: number;
}
