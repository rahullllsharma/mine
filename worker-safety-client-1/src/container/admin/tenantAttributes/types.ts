type SelectedField = {
  /**
   * Keys used to be able to update the BE
   */
  entityKey?: string;
  attributeKey?: string;
  /**
   * Used for the title in the modal
   */
  label: string;
};

type ValidKeys = "visible" | "required" | "filterable";
type FormInputsOptions = {
  key: ValidKeys;
  value: boolean;
  isDisabled: boolean;
};
type FormInputs = {
  label: string;
  labelPlural: string;
  mappings?: Record<string, string[]>;
  mandatory: boolean;
  options: FormInputsOptions[];
};

export type { SelectedField, FormInputs, FormInputsOptions, ValidKeys };
