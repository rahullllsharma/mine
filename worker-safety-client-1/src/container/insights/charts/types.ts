import type { ProjectFiltersPayload } from "@/container/insights/projectFilters/ProjectFilters";
import type { PortfolioFiltersPayload } from "@/container/insights/portfolioFilters/PortfolioFilters";
import type { FiltersDescriptionsReturn } from "./hooks/useFiltersDescriptions";

export enum InsightsTab {
  PORTFOLIO = "PORTFOLIO",
  PROJECT = "PROJECT",
}

export enum InsightsMode {
  PLANNING = "PLANNING",
  LEARNINGS = "LEARNINGS",
}

export type FiltersPayload = ProjectFiltersPayload | PortfolioFiltersPayload;

export type RiskHeatmapProps = {
  filters?: FiltersPayload;
  filtersDescriptions?: FiltersDescriptionsReturn;
  setHasData?: (hasData: boolean) => void;
};

export type PolymorphicRiskHeatmapProps = RiskHeatmapProps & {
  tab: InsightsTab;
};
