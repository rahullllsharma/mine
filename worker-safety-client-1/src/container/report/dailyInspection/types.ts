import type { Project } from "@/types/project/Project";
import type { Location } from "@/types/project/Location";
import type { DailyReport as DailyInspectionReportType } from "@/types/report/DailyReport";
import type { MultiStepFormStep } from "../multiStepForm/state/reducer";
import type { JobHazardAnalysisGraphQLPayloadParams as JobHazardAnalysisFields } from "../jobHazardAnalysis/transformers/graphQLPayload";

import type { jobHazardAnalysisFormInputPrefix } from "@/components/report/jobReport/JobReportCard";
import type { taskFormInputPrefix } from "../tasks/Tasks";
import type {
  workScheduleFormInputPrefix,
  WorkScheduleProps as WorkScheduleFields,
} from "../workSchedule/WorkSchedule";
import type { DailyReportRecommendations } from "@/types/report/DailyReportRecommendations";
import type { FormViewTabStates } from "@/components/forms/Utils";

type DailyInspectionReport = {
  project: Project;
  location: Location;
  dailyReport?: DailyInspectionReportType;
  projectSummaryViewDate?: string;
  recommendations: DailyReportRecommendations["dailyReport"] | null;
  selectedTeb?: FormViewTabStates;
  setSelectedTab?: (tab: string) => void;
};

export type CriticalActivityType = {
  date: string;
  isCritical: boolean;
};
type DailyInspectionReportStepProps = Omit<
  DailyInspectionReport,
  "projectSummaryViewDate"
> & {
  defaults: Required<Pick<DailyInspectionReport, "projectSummaryViewDate">>;
};

type DailyInspectionReportMultiStep = MultiStepFormStep<{
  transformFormDataToSchema?: (
    formData: any,
    options: Pick<DailyInspectionReport, "location">
  ) => any;
}>;

type DailyInspectionReportForms = {
  [workScheduleFormInputPrefix]?: WorkScheduleFields;
  [taskFormInputPrefix]?: Record<string, true>;
  [jobHazardAnalysisFormInputPrefix]?: JobHazardAnalysisFields["jobHazardAnalysis"];
  safetyAndCompliance?: Record<string, never>;
  crew?: Record<string, never> | null;
  additionalInformation: Record<string, never> | null;
};

type DailyInspectionReportSectionKeys = keyof NonNullable<
  DailyInspectionReportType["sections"]
>;

export type {
  DailyInspectionReport,
  DailyInspectionReportStepProps,
  DailyInspectionReportMultiStep,
  DailyInspectionReportForms,
  DailyInspectionReportSectionKeys,
};
