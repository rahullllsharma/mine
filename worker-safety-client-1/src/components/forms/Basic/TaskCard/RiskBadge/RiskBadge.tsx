import { Badge } from "@urbint/silica";
import classnames from "classnames";
import { RiskLevel } from "@/api/generated/types";
import {
  getBackgroundColorByRiskLevel,
  getDefaultTextColorByRiskLevel,
  getRiskIcon,
} from "../utils";

const RiskBadgeAnimated = ({ label }: { label: string }): JSX.Element => {
  return (
    <div
      className="flex items-center font-bold shadow-5 h-5 rounded-full px-1.5 py-0.5 w-max uppercase whitespace-nowrap bg-risk-recalculating text-neutral-shade-75"
      role="note"
    >
      <span className="text-tiny m-1">{label}</span>
    </div>
  );
};

export type RiskBadgeProps = {
  risk: RiskLevel;
  label: string;
};

function RiskBadge({ risk, label }: RiskBadgeProps) {
  if (risk === RiskLevel.Recalculating) {
    return <RiskBadgeAnimated label={label?.toUpperCase()} />;
  }

  return (
    <Badge
      className={classnames(
        "whitespace-nowrap",
        getBackgroundColorByRiskLevel(risk),
        getDefaultTextColorByRiskLevel(risk)
      )}
      iconStart={getRiskIcon(risk)}
      label={label?.toUpperCase()}
    />
  );
}

export { RiskBadge };
