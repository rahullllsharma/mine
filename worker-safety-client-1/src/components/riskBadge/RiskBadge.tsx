import { Badge, Icon } from "@urbint/silica";
import cx from "classnames";
import {
  getBackgroundColorByRiskLevel,
  getDefaultTextColorByRiskLevel,
  getRiskIcon,
} from "@/utils/risk/";
import { RiskLevel } from "./RiskLevel";

interface RiskBadgeAnimatedProps {
  label: string;
  isCritical?: boolean;
}

const RiskBadgeAnimated = ({
  label,
  isCritical = false,
}: RiskBadgeAnimatedProps): JSX.Element => {
  return (
    <div
      className="flex items-center font-bold shadow-5 h-5 rounded-full px-1.5 py-0.5 w-max uppercase whitespace-nowrap bg-risk-recalculating text-neutral-shade-75"
      role="note"
    >
      {/*<div
        className="animate-spin inline-block border-2 border-r-neutral-shade-75 w-3 h-3 mr-1 rounded-full border-neutral-shade-18"
        role="status"
      />*/}
      <span className="text-tiny m-1">{label}</span>
      {isCritical && (
        <Icon
          name={"warning"}
          className={cx(
            "pointer-events-none text-lg bg-transparent ml-1 leading-none"
          )}
        />
      )}
    </div>
  );
};

export interface RiskBadgeProps {
  risk: RiskLevel;
  label: string;
  isCritical?: boolean;
}

export default function RiskBadge({
  risk,
  label,
  isCritical = false,
}: RiskBadgeProps): JSX.Element {
  if (risk === RiskLevel.RECALCULATING) {
    return (
      <RiskBadgeAnimated label={label?.toUpperCase()} isCritical={isCritical} />
    );
  }

  return (
    <Badge
      className={cx(
        "whitespace-nowrap",
        getBackgroundColorByRiskLevel(risk),
        getDefaultTextColorByRiskLevel(risk)
      )}
      iconStart={getRiskIcon(risk)}
      label={label?.toUpperCase()}
      iconEnd={isCritical ? "warning" : undefined}
    />
  );
}
