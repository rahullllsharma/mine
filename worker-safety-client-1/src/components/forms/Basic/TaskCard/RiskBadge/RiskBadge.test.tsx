import { render, screen } from "@testing-library/react";
import { RiskLevel } from "@/api/generated/types";

import { RiskBadge } from "./RiskBadge";

describe(RiskBadge.name, () => {
  const props = (risk: RiskLevel) => ({
    risk,
    label: `${risk} risk`,
  });

  it("renders properly the high risk badge", () => {
    const { asFragment } = render(<RiskBadge {...props(RiskLevel.High)} />);

    expect(screen.getByText("HIGH RISK"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders properly the medium risk badge", () => {
    const { asFragment } = render(<RiskBadge {...props(RiskLevel.Medium)} />);

    expect(screen.getByText("MEDIUM RISK"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders properly the low risk badge", () => {
    const { asFragment } = render(<RiskBadge {...props(RiskLevel.Low)} />);

    expect(screen.getByText("LOW RISK"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders properly the unknown risk badge", () => {
    const { asFragment } = render(<RiskBadge {...props(RiskLevel.Unknown)} />);

    expect(screen.getByText("UNKNOWN RISK"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders properly the recalculating risk badge", () => {
    const { asFragment } = render(
      <RiskBadge {...props(RiskLevel.Recalculating)} />
    );

    expect(screen.getByText("RECALCULATING RISK"));
    expect(asFragment()).toMatchSnapshot();
  });
});
