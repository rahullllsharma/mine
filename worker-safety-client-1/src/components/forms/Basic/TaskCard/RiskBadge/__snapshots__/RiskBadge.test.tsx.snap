// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiskBadge renders properly the high risk badge 1`] = `
<DocumentFragment>
  <div
    class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-high text-white"
    role="note"
  >
    <span
      class="flex items-center px-1"
    >
      <i
        aria-hidden="true"
        class="ci-chevron_up"
      />
      <span
        class="text-tiny mr-1"
      >
        HIGH RISK
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`RiskBadge renders properly the low risk badge 1`] = `
<DocumentFragment>
  <div
    class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-low text-white"
    role="note"
  >
    <span
      class="flex items-center px-1"
    >
      <i
        aria-hidden="true"
        class="ci-chevron_down"
      />
      <span
        class="text-tiny mr-1"
      >
        LOW RISK
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`RiskBadge renders properly the medium risk badge 1`] = `
<DocumentFragment>
  <div
    class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-medium text-neutral-shade-100"
    role="note"
  >
    <span
      class="flex items-center px-1"
    >
      <i
        aria-hidden="true"
        class=""
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.74 11.921c-.497-.053-.91.09-1.21.441-.364.428-.53.811-.53 1.138H6c0-.998.492-1.83 1.008-2.435.796-.934 1.902-1.245 2.947-1.132 1.021.11 2.01.619 2.752 1.36.453.453 1.03.73 1.552.786.498.053.91-.09 1.21-.441.365-.428.531-.812.531-1.138h2c0 .998-.492 1.83-1.008 2.435-.796.934-1.902 1.245-2.947 1.132-1.021-.11-2.01-.619-2.752-1.36-.453-.453-1.03-.73-1.552-.786z"
            fill-rule="evenodd"
          />
        </svg>
      </i>
      <span
        class="text-tiny mr-1"
      >
        MEDIUM RISK
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`RiskBadge renders properly the recalculating risk badge 1`] = `
<DocumentFragment>
  <div
    class="flex items-center font-bold shadow-5 h-5 rounded-full px-1.5 py-0.5 w-max uppercase whitespace-nowrap bg-risk-recalculating text-neutral-shade-75"
    role="note"
  >
    <span
      class="text-tiny m-1"
    >
      RECALCULATING RISK
    </span>
  </div>
</DocumentFragment>
`;

exports[`RiskBadge renders properly the unknown risk badge 1`] = `
<DocumentFragment>
  <div
    class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-unknown text-neutral-shade-75"
    role="note"
  >
    <span
      class="flex items-center px-1"
    >
      <span
        class="text-tiny mx-1"
      >
        UNKNOWN RISK
      </span>
    </span>
  </div>
</DocumentFragment>
`;
